src/pages/Home/home.js

```diff
@@ -34,61 +32,24 @@ export class Home extends React.Component {
     this.getBookData("history");
   }
 
-  mappedData = (genre) => {
-    if (this.state[genre] && !this.state.isLoading) {
-      const mappedData = this.state[genre].map((book) => {
-        return (
-          <Card>
-            <Link
-              to={`/book/${book.volumeInfo.industryIdentifiers[0].identifier}`}
-            >
-              <div>{book.volumeInfo.title}</div>
-              <div>
-                <img
-                  src={book.volumeInfo.imageLinks.thumbnail}
-                  alt={book.volumeInfo.title}
-                />
-              </div>
-            </Link>
-          </Card>
-        );
-      });
-      return <Container>{mappedData}</Container>;
-    }
-  };
-
   render() {
     return (
       <>
         {this.state.isLoading && <>Loading...</>}
-        {!this.state.isLoading && (
-          <>
-            <>
-              <h1>Horror</h1>
-              <div>{this.mappedData("horror")}</div>
-            </>
-
-            <>
-              <h1>Romance</h1>
-              <div>{this.mappedData("romance")}</div>
-            </>
-
-            <>
-              <h1>Mystery</h1>
-              <div>{this.mappedData("mystery")}</div>
-            </>
-
-            <>
-              <h1>Nonfiction</h1>
-              <div>{this.mappedData("nonfiction")}</div>
-            </>
-
+        {!this.state.isLoading &&
+          this.state.horror &&
+          this.state.romance &&
+          this.state.mystery &&
+          this.state.nonfiction &&
+          this.state.history && (
```

Comment:
wow can this be done in a better way? how could you encapsulate this into a function?